scipy/fft/_basic_backend.py

```diff
@@ -32,6 +35,13 @@ def _execute_1D(func_str, pocketfft_func, x, n, axis, norm, overwrite_x, workers
     norm = _validate_fft_args(workers, plan, norm)
     if hasattr(xp, 'fft'):
         xp_func = getattr(xp.fft, func_str)
+        if func_str in complex_funcs:
+            try:
+                res = xp_func(x, n=n, axis=axis, norm=norm)
+            except TypeError: # backends may require complex input
```

Comment:
There is no guarantee that we get a `TypeError` here, so maybe this would be better as a blanket `except`?